src/routes/safe/components/Transactions/TxsTable/columns.js

```diff
@@ -34,29 +33,40 @@ type TxData = {
 
 export const formatDate = (date: string): string => format(parseISO(date), 'MMM d, yyyy - HH:mm:ss')
 
-export const getIncomingTxAmount = (tx: IncomingTransaction) => {
-  const txAmount = tx.value ? `${new BigNumber(tx.value).div(`1e${tx.decimals}`).toFixed()}` : 'n/a'
-  return `${txAmount} ${tx.symbol || 'n/a'}`
+type TxValues = {
+  value?: string | number,
+  decimals?: string | number,
+  symbol?: string,
 }
 
-export const getTxAmount = (tx: Transaction) => {
-  const web3 = getWeb3()
-  const { fromWei, toBN } = web3.utils
-
-  let txAmount = 'n/a'
-
-  if (tx.isTokenTransfer && tx.decodedParams) {
-    const tokenDecimals = tx.decimals.toNumber ? tx.decimals.toNumber() : tx.decimals
-    txAmount = `${formatAmount(
-      BigNumber(tx.decodedParams.value)
-        .div(10 ** tokenDecimals)
-        .toString(),
-    )} ${tx.symbol}`
-  } else if (Number(tx.value) > 0) {
-    txAmount = `${fromWei(toBN(tx.value), 'ether')} ${tx.symbol}`
+const NOT_AVAILABLE = 'n/a'
+
+const getAmountWithSymbol = ({ decimals = 0, symbol = NOT_AVAILABLE, value }: TxValues, formatted = false) => {
+  const nonFormattedValue = BigNumber(value).times(`1e-${decimals}`).toFixed()
+  const finalValue = formatted ? formatAmount(nonFormattedValue).toString() : nonFormattedValue
```

Comment:
`toString()` is not required since formatAmount returns a string

Reply:
actually, when you pass something that's not a number `formatAmount` returns a `NaN` object and `bigNumber` returns `'NaN'`, so I'm asking for its string representation so it simplifies the next line: 

https://github.com/gnosis/safe-react/blob/feature/tx-balance-notations/src/routes/safe/components/Transactions/TxsTable/columns.js#L47

Reply:
👍